agent/provider/src/provider_agent.rs

```diff
@@ -172,14 +172,33 @@ impl GlobalsState {
 impl ProviderAgent {
     pub async fn new(mut args: RunConfig, config: ProviderConfig) -> anyhow::Result<ProviderAgent> {
         let data_dir = config.data_dir.get_or_create()?;
-        let log_handler = start_logger("info", Some(&data_dir), &vec![])?;
+
+        //log_dir is the same as data_dir by default, but can be changed using --log-dir option
+        let log_dir = if let Some(log_dir) = &config.log_dir {
+            log_dir.get_or_create()?
+        } else {
+            data_dir.clone()
+        };
+
+        //if --debug option is provided override RUST_LOG flag with debug defaults
+        //if you want to more detailed control over logs use RUST_LOG variable and do not use --debug flag
+        if args.debug {
+            std::env::set_var(
+                "RUST_LOG",
+                "debug,tokio_core=info,tokio_reactor=info,hyper=info,reqwest=info",
+            );
+        }
+        //start_logger is using env var RUST_LOG internally
+        let log_handler = start_logger("info", Some(&log_dir), &vec![])?;
```

Comment:
You shouldn't use RUST_LOG variable. Check what `start_logger` function is able to do.
In first parameter you can pass log level. Last parameter is list of modules to filter

Reply:
start_logger is using RUST_LOG internally and it is ignoring everything else if it set. If someone uses --debug flag I think we should override it.